feature/spi-driver

nada-ehab opened the pull request:
implementing spi_INIT function that initiates master or slave mode , enables SPI diver and the interrupt and determines data order LSB pr MSB

nada-ehab committed:
implement(spi): implement init function

implementing spi_INIT function that initiates master or slave mode , enables SPI diver and the interrupt and determines data order LSB pr MSB

nada-ehab committed:
set(spi): clock set function

setting clock priority, clock phase and clock rate

hussein-shamy committed:
main merge

hussein-shamy committed:
main merge

hussein-shamy commented:
The driver is not complete. Kindly complete it by adding the missing functionalities.

nada-ehab committed:
implement(spi): implement byte transfer function

he function performs an SPI byte transfer, waits for completion or timeout, and handles the state of the SPI interface, returning an appropriate status

nada-ehab committed:
implement (spi): implement TranceiverSynch

SPI_BufferTranceiverSynch function synchronously transmits and receives a specified number of bytes (buffer size) over SPI by using the SPI_ uint8_tTranceive function in a loop.

nada-ehab committed:
implement(spi)SPI_u8BufferTranceiverAsync

Asynchronously transmits and receives a buffer of data via SPI.
  spi_buffer  Pointer to a structure containing the buffer and other relevant settings for asynchronous SPI communication.
 The status of the SPI operation.

hussein-shamy committed:
fix(spi): remove the conflicts

hussein-shamy committed:
Merge branch 'main' into feature/spi-driver

hussein-shamy committed:
fix(spi): fix some issues

hussein-shamy committed:
fix(spi): fix some issues
